### PR TITLE
crosscluster: store LDR job IDs in table descriptor

### LINKED ISSUE #129728: catalog: add LDRJobIDs to table descriptor with an API to add/remove them
See https://docs.google.com/document/d/1Wj-6lu7N4p3vPGJ9qDPBbu9Vu8haTpcZ4Z5cKzYTE7c/edit

Table descriptors should have a field to keep track of all the logical data replication job IDs that rely on that table.

This field will be used to restrict certain schema changes on those tables.

Epic CRDB-38974
Jira issue: CRDB-41693

### PR DESCRIPTION
**Pull request by rafiss**
This adds a new field to table descriptors that will keep track of LDR
jobs that are referencing this table as a destination.

This slice will be used in a future change to block schema changes on
this table if the slice is non-empty.

fixes https://github.com/cockroachdb/cockroach/issues/129728
Release note: None

### CODE REVIEW

File: pkg/sql/catalog/tabledesc/table_desc_test.go [hunk 1]
```diff
@@ -150,6 +150,39 @@ func TestStripDanglingBackReferencesAndRoles(t *testing.T) {
 			strippedDanglingBackReferences: true,
 			strippedNonExistentRoles:       true,
 		},
+		{
+			name: "LDR job IDs",
+			input: descpb.TableDescriptor{
+				Name: "foo",
+				ID:   104,
+				MutationJobs: []descpb.TableDescriptor_MutationJob{
```
Comment (msbutler) [inline pkg/sql/catalog/tabledesc/table_desc_test.go:158]: this test case should not require mutation jobs, right?

**Review by msbutler — APPROVED** [review]

File: pkg/sql/catalog/tabledesc/table_desc_test.go (hunk 1 above)
Comment (rafiss) [reply pkg/sql/catalog/tabledesc/table_desc_test.go:158]: that's right; but just wanted to have a test that cleans up both types of jobs on the same descriptor

**Comment by rafiss** [comment]
tftr!

bors r+

**Comment by msbutler** [comment]
i think this will conflict slightly with https://github.com/cockroachdb/cockroach/pull/130592

**Comment by msbutler** [comment]
oops, didn't conflict.

bors r+

**Pull request merged**